packages/ui/src/components/Schemas/BarcodeSchema.tsx

```diff
@@ -54,18 +35,44 @@ const ErrorBarcode = () => (
 );
 
 const BarcodePreview = (props: { schema: BarcodeSchema; value: string }) => {
+  const [imageSrc, setImageSrc] = useState<string>('');
+  const [isBarcodeValid, setIsBarcodeValid] = useState<boolean>(true);
+
   const { schema, value } = props;
+  const { type, height, width, barcolor, backgroundcolor, textcolor } = schema;
 
-  if (value.length === 0) {
-    return null;
-  }
+  useEffect(() => {
+    const canvas = document.createElement('canvas');
+    const barcodeType = type === 'nw7' ? 'rationalizedCodabar' : type;
 
-  return validateBarcodeInput(schema.type as BarCodeType, value) ? (
-    <SampleBarcode schema={schema} />
+    try {
+      bwipjs.toCanvas(canvas, {
```

Comment:
Real time preview barcode! i like this.

Reply:
Wait I guess we can extract this creating barcode image process to common package.

I wrote same code in generator package

Reply:
I have looked at this code and I couldn't easily created a single function here. 
The existing `createBarCode` function returns a `Promise` to a `Buffer` whereas this new code uses the canvas directly. It's also not doing much beyond creating the parameters for the function call.

So instead I have moved some of the logic that creates these parameters into shared functions that both the UI and generator use.

Let me know what you think.

Reply:
memo:

```
function bufferToDataURL(blob) {
    return new Promise((resolve, reject) => {
        const reader = new FileReader();
        reader.onloadend = () => resolve(reader.result);
        reader.onerror = reject;
        reader.readAsDataURL(blob);
    });
}

const buffer = new Blob([new Uint8Array([/* binary data */])], { type: 'image/png' });
bufferToDataURL(buffer).then(dataURL => {
    console.log(dataURL);
}).catch(error => {
    console.error('Error:', error);
});
```

Reply:
I successfully moved createBarCode to common.

ref: https://github.com/pdfme/pdfme/commit/8d4b9f00f2fe3ef8f9d44c703f45bdac039fd241